backend/server/server_utils.py

```
@@ -103,16 +116,10 @@ def sanitize_filename(filename: str) -> str:
     # Split into components
     prefix, timestamp, *task_parts = filename.split('_')
     task = '_'.join(task_parts)
-    
-    # Calculate max length for task portion
-    # 255 - len(os.getcwd()) - len("\\gpt-researcher\\outputs\\") - len("task_") - len(timestamp) - len("_.json") - safety_margin
-    max_task_length = 255 - len(os.getcwd()) - 24 - 5 - 10 - 6 - 5  # ~189 chars for task
-    
-    # Truncate task if needed
-    truncated_task = task[:max_task_length] if len(task) > max_task_length else task
-    
+    task_hash = hashlib.md5(task.encode('utf-8', errors='ignore')).hexdigest()[:10]
```

Ruff (0.15.2)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)